### PR TITLE
[SNOW-980540] Fix for sql simplifier that incorrectly pushed filter down during flattern with window function

### PR DESCRIPTION
**Pull request by sfc-gh-yzou**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.
SNOW-980540

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.
The sql simplifier incorrectly flattened the following query 
```
  FROM ( 
    SELECT "A", "B", min("B") OVER (  ) AS "C" 
      FROM ( 
        SELECT "A", iff(("A" = 0 :: INT), 10, 11) AS "B" 
        FROM ( 
          SELECT "A" FROM ( 
            SELECT $1 AS "A" FROM  VALUES (0 :: INT), (1 :: INT))))) 
            WHERE ("A" = 1 :: INT)
```
into
```
SELECT 
  "A", "B", min("B") OVER (  ) AS "C" 
    FROM ( 
      SELECT "A", iff(("A" = 0 :: INT), 10, 11) AS "B" 
      FROM ( 
        SELECT $1 AS "A" 
        FROM  VALUES (0 :: INT), (1 :: INT))) WHERE ("A" = 1 :: INT)
```
where the filter got pushed into the flattend statement.

### CODE REVIEW

**Comment by sfc-gh-yzou** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/snowflake/snowpark/mock/_plan.py [hunk 1]
```diff
@@ -120,7 +120,10 @@
     Sample,
 )
 from snowflake.snowpark._internal.type_utils import infer_type
-from snowflake.snowpark._internal.utils import generate_random_alphanumeric, parse_table_name
```
Comment (sfc-gh-yzou) [inline src/snowflake/snowpark/mock/_plan.py:123]: not related, updated automatically by precommit

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## 1.11.2 (TBD)
+
```
Comment (sfc-gh-aalam) [inline CHANGELOG.md:4]:
```suggestion
## 1.12.0 (TBD)

```

**Review by sfc-gh-yixie — APPROVED** [review]

**Review by sfc-gh-aalam — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (sfc-gh-yzou) [reply CHANGELOG.md]: fixed

**Review by sfc-gh-stan — APPROVED** [review]

**Pull request merged**